src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -298,6 +371,34 @@ private static void ChangeColor(bool foreground, ConsoleColor color)
         /// <summary>Cache of the format strings for foreground/background and ConsoleColor.</summary>
         private static readonly string[,] s_fgbgAndColorStrings = new string[2, 16]; // 2 == fg vs bg, 16 == ConsoleColor values
 
+        public static bool TryGetSpecialConsoleKey(char[] givenChars, int startIndex, int endIndex, out ConsoleKey key, out int keyLength)
+        {
+            key = (ConsoleKey)0;
+            keyLength = 0;
+            int unprocessedCharCount = endIndex - startIndex + 1;
+
+            if (unprocessedCharCount < TerminalKeyInfo.Instance.MinKeyLength)
```

Comment:
> how will we differentiate

Why would we need to?  If we poll and we can populate more data into the buffer, then we do and we continue processing.  If we poll and we can't get more data, then we know we didn't get a partial key due to previously having too full a buffer.